schema/deploy/trigger_functions/create_or_refresh_review_step.sql

```diff
@@ -0,0 +1,47 @@
+-- Deploy ggircs-portal:trigger_functions/create_or_refresh_review_step to pg
+-- requires: tables/review_step
+
+begin;
+
+create or replace function ggircs_portal_private.create_or_refresh_review_step()
+  returns trigger as $$
+
+declare
+  temp_row record;
+
+begin
+
+  for temp_row in
+    select id, step_name from ggircs_portal.review_step_name where is_active = true
+  loop
+    if exists(select * from ggircs_portal.review_step where application_id = new.application_id and review_step_name_id = temp_row.id) then
+      update ggircs_portal.review_step set is_complete = false where application_id = new.application_id and review_step_name_id = temp_row.id;
+    else
+      insert into ggircs_portal.review_step(application_id, review_step_name_id)
+        values (new.application_id, temp_row.id);
+    end if;
+  end loop;
+
```

Comment:
Wild suggestion, this might be completely out of scope:
We could use an upsert instead if we create a unique index on both columns - this would have the benefit of guaranteeing that we only have at most one instance of each review_step_name per application.

```SQL
create unique index ggircs_portal_review_step_on_app_id_and_review_step_name_id on ggircs_portal.review_step(application_id, review_step_name_id);

insert into ggircs_portal.review_step(application_id, review_step_name_id)
        values (new.application_id, temp_row.id)
        on conflict (application_id, review_step_name_id)
        do update set is_complete=false;